packages/cypress/src/reporter.ts

```diff
@@ -118,19 +118,23 @@ class CypressReporter {
             path: [spec.relative],
             result: "unknown",
             relativePath: spec.relative,
+            steps: [],
           },
         ],
       };
     }
 
     let testsWithSteps: Test[] = [];
     let hooksWithSteps: Hook[] = [];
+    let stepsError: string | undefined;
```

Comment:
So there would now be both an optional `error` attribute _and_ an optional `stepsError` attribute – both describing why a given test failed? That seems kind of confusing to me. What if both are present?

I prefer to avoid ambiguous data situations if possible. I guess how "okay" I am with this depends on how temporary it is. If you think we'll be able to get to SCS-1023 soon then it doesn't matter too much. Else I think I might want to re-think this design to incorporate a new `result` enum and then use a single` error` attribute.

(For that matter, I'd like the `error` attribute to _always_ be the thing we use when showing a top-level failure. Right now we have to scan steps for an error first, because it has more details– but I guess that's getting side tracked.)

Reply:
It's not ideal but those error fields are referring to different things. `error` means the test failed and `stepsError` means the plugin had a problem. Either could be true without the other. IOW, a test can pass but we fail to build the step metadata so `stepsError` would be valued.

Reply:
I realize that you're proposing they refer to different things, but I think that's an awkward data shape to work with.

It's particularly awkward naming, IMO, since test have "steps" which might have "errors" and "error" is more closely related to a `steps.error` than `stepsError` is.

Reply:
I think this is the best we can do without changing the overall structure. If you'd rather wait for that, that's fine with me

Reply:
I'm okay with us taking an incremental step, so long as we _do_ address the overall structure at some point (preferably not super far out) – but I don't think `stepsError` is the best name, even for an incremental step.

Maybe something like– `pluginError` or `integrationError` ? So it's clearly _separate_ from any errors that may have happened in one or more of the test steps.